### PR TITLE
add support for retreiving value of received temporary key in server …

### PR DESCRIPTION
**Pull request by zeevt**
…key exchange message, to implement functionality like openssl's ssl_print_tmp_key

### CODE REVIEW

**Comment by alex** [comment]
ok to test

File: src/cryptography/hazmat/bindings/openssl/ec.py [hunk 1]
```diff
@@ -188,6 +189,8 @@
 const EC_METHOD *EC_GF2m_simple_method();
 
 int EC_METHOD_get_field_type(const EC_METHOD *);
+
+const char *EC_curve_nid2nist(int nid);
```
Comment (alex) [inline src/cryptography/hazmat/bindings/openssl/ec.py]: The names of parameters should not be included, meaning this should be written `const char *EC_curve_nid2nist(int);`

(This applies on a few other lines)

File: src/cryptography/hazmat/bindings/openssl/ssl.py [hunk 2]
```diff
@@ -591,6 +595,18 @@
 static const long Cryptography_HAS_COMPRESSION = 1;
 #endif
 
+#if defined(SSL_CTRL_GET_SERVER_TMP_KEY)
+static const long Cryptography_HAS_GET_SERVER_TMP_KEY = 1;
+#undef SSL_get_server_tmp_key
+long SSL_get_server_tmp_key(SSL *ssl, EVP_PKEY **key) {
+    return SSL_ctrl(ssl, SSL_CTRL_GET_SERVER_TMP_KEY, 0, key);
+}
```
Comment (alex) [inline src/cryptography/hazmat/bindings/openssl/ssl.py]: Why are you redefining the macro? You should just be able to bind it, same as the other macros.

**Comment by zeevt** [comment]
Oh, it's building again before I was ready. Still working on fixing libressl compatibility.

**Comment by reaperhulk** [comment]
Any reason you want nid2nist over `OBJ_nid2sn`? (Example of that being used to get the curve name here: https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/ec.py#L43)

I'm fine with binding this, just curious.

**Comment by zeevt** [comment]
@reaperhulk: I've implemented openssl's `ssl_print_tmp_key`[1] for pyopenssl and want cryptography to have all the symbols (if openssl 1.0.2 is used). The SECG name of the curve is enough, the NIST name is not strictly necessary, but openssl thought it would be nice to have, so I'd like to have it.

1 - https://git.openssl.org/gitweb/?p=openssl.git;a=blob;f=apps/s_cb.c;h=ee57537083afd3714f14300ec9532691cbb288ca;hb=refs/heads/OpenSSL_1_0_2-stable#l506

**Comment by reaperhulk** [comment]
Sounds reasonable :)

**Pull request merged**